src/translator/text_processor.cpp

```diff
@@ -3,27 +3,85 @@
 #include <vector>
 
 #include "annotation.h"
+#include "common/cli_helper.h"
 #include "common/options.h"
 #include "data/types.h"
 #include "definitions.h"
 
 namespace marian {
 namespace bergamot {
 
+namespace {
+ug::ssplit::SentenceStream::splitmode string2splitmode(const std::string &m) {
+  typedef ug::ssplit::SentenceStream::splitmode splitmode;
+  if (m == "sentence") {
+    return splitmode::one_sentence_per_line;
+  } else if (m == "paragraph") {
+    return splitmode::one_paragraph_per_line;
+  } else if (m == "wrapped_text") {
+    return splitmode::wrapped_text;
+  } else {
+    ABORT("Unknown ssplitmode {}, Please choose one of {sentence,paragraph,wrapped_text}");
+  }
+}
+
+ug::ssplit::SentenceSplitter loadSplitter(const std::string &ssplitPrefixFile) {
+  // Temporarily supports empty, will be removed when mozilla passes ssplitPrefixFile
+  ug::ssplit::SentenceSplitter splitter;
+  if (ssplitPrefixFile.size()) {
+    std::string interpSsplitPrefixFile = marian::cli::interpolateEnvVars(ssplitPrefixFile);
+    LOG(info, "Loading protected prefixes for sentence splitting from {}", interpSsplitPrefixFile);
+    splitter.load(interpSsplitPrefixFile);
+  } else {
+    LOG(warn,
+        "Missing list of protected prefixes for sentence splitting. "
+        "Set with --ssplit-prefix-file.");
+  }
+  return splitter;
+}
+
+ug::ssplit::SentenceSplitter loadSplitter(const AlignedMemory &memory) {
+  // Temporarily supports empty, will be removed when mozilla passes memory
+  ug::ssplit::SentenceSplitter splitter;
+  if (memory.size()) {
+    std::string_view serialized(memory.begin(), memory.size());
+    splitter.loadFromSerialized(serialized);
+  }
+  return splitter;
+}
+
+}  // namespace
+
 Segment TextProcessor::tokenize(const string_view &segment, std::vector<string_view> &wordRanges) {
   // vocabs_->sources().front() is invoked as we currently only support one source vocab
   return vocabs_.sources().front()->encodeWithByteRanges(segment, wordRanges, /*addEOS=*/false, /*inference=*/true);
 }
 
-TextProcessor::TextProcessor(Vocabs &vocabs, Ptr<Options> options) : vocabs_(vocabs), sentence_splitter_(options) {
-  max_length_break_ = options->get<int>("max-length-break");
-  max_length_break_ = max_length_break_ - 1;
-  ABORT_IF(max_length_break_ < 0, "max-length-break cannot be < 0");
+TextProcessor::TextProcessor(Ptr<Options> options, const Vocabs &vocabs, const std::string &ssplit_prefix_file)
```

Comment:
Do we preserver an option for no ssplit file provided?

Reply:
Yeah, I put it back in, although it's not tested because common args are modified to include ssplit-prefix-file, effecting across all tests (https://github.com/browsermt/bergamot-translator-tests/pull/37). Should I forbid this behaviour? I'm inclined towards (1) bifurcating ByteArray based construction and FilePath based construction into two explicit paths and additionally constraining that (2) all these payloads *should* exist for bergamot-translator (i.e ssplit file should always be provided). 

Checking if ByteArray Empty to then switch-over to filepath based load to then a case of no-prefix file seems rather unpleasant.

Reply:
My concern is that the models we have currently don't include ssplit prefix files. Also, mozilla have their own fork of the models. We would need to negotiate ssplit prefix being mandatory with everyone before merging this.

Reply:
Well, let's add the ssplit files to the models.